cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +356,38 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubeadmVersion    string
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+
+	// Skip this check for "super-custom builds", where apimachinery/the overall codebase version is not set.
+	if kubever.KubeadmVersion == "v0.0.0" {
+		return nil, nil
+	}
+
+	kadmVersion, err := semver.ParseTolerant(kubever.KubeadmVersion)
```

Comment:
NB. To implements this I anticipated in this PR the change on `pkg/util/version` already present in #48899 (https://github.com/kubernetes/kubernetes/pull/48899/files#diff-04d254facbc5683bce0e9aba7cad624f)